### DISCUSSION TITLE
isSubmitSuccessful true on Promise.reject

### DISCUSSION THREAD

**Discussion by borm** [react-hook-form/react-hook-form#3949] [Ideas] [answered]
```
function CustomForm 
  const { ...., formState } = useForm();
  const { isSubmitSuccessful } = formState;
  const onSubmit = async (data) => {
    console.log(data);
    return Promise.reject("Error");
  };

  console.log(isSubmitSuccessful) // true ??? why
```

https://codesandbox.io/s/react-hook-form-get-started-forked-9s3li?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
hook form only takes care of the form validation, not the promise.

**Reply by borm** [reply]
but i really need to know what my submission was successful
please look at final-form, when u return promise reject is not success

**Reply by bluebill1049** [reply]
could you just do a setState after the promise?

**Reply by bluebill1049** [reply]
i mean isSubmitSuccesful is at the form level, not the result level. you know what I mean right?

**Reply by bluebill1049** [reply]
yea, we don't have that feature at the moment.

**Reply by borm** [reply]
yep, but, i no need use additional state for know that

**Reply by borm** [reply]
thats will be a good feature

**Reply by borm** [reply]
is a real case, i got 400 from api, so is not successful (;

**Reply by bluebill1049** [reply]
you want to send a feature request for that? we will put into pending stage and see if others have a similar request.

**Reply by borm** [reply]
yep, thats is really needs, so look at difference

1.
```
const onSubmit = (data: MyFormData) => {
  try {
     await API.post(....)
  }   catch(error) { return Promise.reject(error) }
}
```

and 
2.
```
const onSubmit = (data: MyFormData) => API.post(....)
// here we can understand then or catch
```

and
3.
```
const [isSuccess, setIsSuccess] = useState(false);
const onSubmit = (data: MyFormData) => {
  try {
     await API.post(....)
  }   catch(error) { setIsSuccess(false) } // useless logic here
}
```

my choose is numba 2 as well)

**Reply by bluebill1049** [reply]
1 or 2 both works

**Reply by borm** [reply]
am, just now no) or i did`nt understood u correctly

**Reply by bluebill1049** [reply]
i mean 1 or 2 both works for feature request.

**Reply by borm** [reply]
https://github.com/react-hook-form/react-hook-form/issues/3956

**Reply by borm** [reply]
and as always thank you so much for a fast replies)
